### PR TITLE
feat: toast

### PR DESCRIPTION
**Pull request by RobChangCA**
# Pull Request Checklist

- [x] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [x] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [x] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [x] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [x] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [x] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

# Description
Adds a toast component to display a success / error when transferring and NFT.
Screenshot 2024-09-27 at 6 04 12 PM

# Testing steps
Attempt to transfer an NFT, on success you should see a green toast appear on top of the white nav bar.
You can click the X next to the toast to dismiss it early.  It will naturally dismiss after 4 seconds.
This toast is not themeable as the nav bar is always white.

---

## PR-Codex overview
This PR introduces a `Toast` notification system using `@radix-ui/react-toast`. It adds a `ToastProvider`, updates the UI components to utilize the toast functionality, and implements animations for toast visibility.

### Detailed summary
- Added `@radix-ui/react-toast` dependency.
- Created `ToastProvider` in `ToastProvider.tsx`.
- Implemented `useToast` hook in `useToast.tsx`.
- Added `Toast` component for displaying notifications.
- Integrated toast notifications in `MintCard` for success/error messages.
- Added toast animations in `globals.css`.
- Created `XIcon` component for closing the toast.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: examples/ui-demo/src/components/shared/MintCard.tsx [hunk 1]
```diff
@@ -96,13 +107,19 @@ export const MintCard = () => {
...
       console.log("uri", uri);
       return uri;
-
     },
     enabled: !!client && !!client?.readContract,
   });
 
   return (
     <div className="flex bg-bg-surface-default radius-1 border-btn-secondary border-2 overflow-hidden h-[532px]">
+      <Toast
```
Comment (rthomare) [inline examples/ui-demo/src/components/shared/MintCard.tsx:116]: Do we plan on using this toast in multiple places. In the way it is currently implemented every component needs to instantiate and open + close their own toast, and handle the logic. 

Can we do better here? Could we wrap this all up in a toast provider component that exposes a useOpenToast function that takes in some parameters simply open and then auto close the toast globally? 

That way when we want to use it for errors, or something else, it's a bit easier for downstream components?

Comment (moldy530) [reply examples/ui-demo/src/components/shared/MintCard.tsx:116]: looks like this is using a ToastProvider already, but it is weird that we can't just render a global Toast component and show a `Toast` via some kind of `showToast(...)` method. 

Maybe we could do something like this? https://www.radix-ui.com/primitives/docs/components/toast#imperative-api

the `useImperativeHandle` hook which contains a `publish` method is interesting here. I think we could use that to be able to:
1. render a single Toast instance in the root of the app
2. add to the context of the ui-demo app a ref or wrapper around a ref to the toast and export a method like `showToast({ content, type })`

Then we can have a `useToast` hook or something that lets us access that `showToast` method

Comment (RobChangCA) [reply examples/ui-demo/src/components/shared/MintCard.tsx:116]: The intention was to have an explicit interface to reduce abstraction, I will refactor to use a hook implementation.

File: examples/ui-demo/src/hooks/useToast.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+"use client";
+
+import { ToastContext } from "@/contexts/ToastProvider";
+import { useContext } from "react";
+
+export const useToast = () => {
+  const { toast, setToast, closeToast } = useContext(ToastContext);
```
Comment (rthomare) [inline examples/ui-demo/src/hooks/useToast.tsx:7]: Can we throw an error if the context comes back as undefined from `useContext`?

Something like 
```
if (!context) { 
   throw Error("useToast can only be used by components that are descendants of the ToastProvider`");
}
```

Comment (rthomare) [reply examples/ui-demo/src/hooks/useToast.tsx:7]: And then you can just return context no need to break it down and then send it a new object.

File: examples/ui-demo/src/components/shared/Toast.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+          <span
+            className={`${
+              type === "success" ? "bg-[#16A34A]" : "bg-[#DC2626]"
+            } align-middle px-2 py-1 text-fg-invert text-xs font-semibold rounded mr-2`}
+          >
+            {type === "success" ? "Success" : "Error"}
+          </span>
+          {text}
+        </p>
+        <Close>
```
Comment (rthomare) [inline examples/ui-demo/src/components/shared/Toast.tsx:64]: Should this be hooked up to an action for closing the toast?

File: examples/ui-demo/src/components/shared/Toast.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  return (
+    <>
+      <Root
+        id="toast"
+        open={open}
+        onOpenChange={closeToast}
+        className={`${getBGColor()} align-middle rounded-lg shadow-lg px-3 py-2 flex justify-center items-center`}
+      >
+        <p className={`${getTextColor()} text-center pr-2 align-middle`}>
+          <span
```
[the remark this answers is not in the document]
Comment (RobChangCA) [reply examples/ui-demo/src/components/shared/Toast.tsx:55]: Great catch, missed that.  Will hard code the value

**Review by rthomare — APPROVED** [review]

**Pull request merged**